api/answer.go

```diff
@@ -32,15 +30,120 @@ var (
-  where    deleted_at is NULL and answers.parent is NULL and answers.question = ? 
+  where    deleted_at is NULL and answers.parent is NULL and answers.question = ?
 `, VOTES_QUERY)
 	REPLIES_QUERY = `
   select   *
   from     answers
   where    deleted_at is NULL and answers.parent IN ?
 `
+	names = []string{"Wyatt", "Vivian", "Maria", "Alexander", "Luis", "Aidan", "Mason", "Aiden", "Mackenzie", "Adrian", "Oliver", "Andrea", "Amaya", "Nolan", "Riley", "Robert", "Ryker", "Sara", "Ryan", "Sawyer"}
 )
 
+func ConvertAnswerToAPI(answer Answer, id uint) (*AnswerResponse, error) {
+	db := util.GetDb()
+	usr, err := GetUserByID(db, id)
+	if err != nil {
+		return nil, err
+	}
+
+	var avatar, username string
+
+	if answer.Anonymous {
+		avatar = util.GenerateAnonymousAvatar(usr.Alias)
+		username = usr.Alias
+	} else {
+		avatar = fmt.Sprintf("https://avatars.githubusercontent.com/u/%d?v=4", usr.ID)
+		username = usr.Username
+	}
+
+	// recursively convert replies
+	var replies []AnswerResponse
+	for _, reply := range answer.Replies {
+		reply, err := ConvertAnswerToAPI(reply, id)
+		if err != nil {
+			return nil, err
+		}
+		replies = append(replies, *reply)
+	}
+
+	return &AnswerResponse{
+		ID:        answer.ID,
+		CreatedAt: answer.CreatedAt,
+		UpdatedAt: answer.UpdatedAt,
+		Question:  answer.Question,
+		Parent:    answer.Parent,
+		User:      username,
+		Content:   answer.Content,
+		Upvotes:   answer.Upvotes,
+		Downvotes: answer.Downvotes,
+		Replies:   replies,
+		Anonymous: answer.Anonymous,
+		AvatarURL: avatar,
+	}, nil
+
+}
+
+func GetUserByID(db *gorm.DB, id uint) (*User, error) {
+	var user User
+	if err := db.First(&user, id).Error; err != nil {
+		return nil, err
+	}
+	return &user, nil
+}
+
+func GetOrCreateUserByID(db *gorm.DB, id uint, username string) (*User, error) {
+	user, err := GetUserByID(db, id)
+	if err == nil {
+		return user, nil
+	}
+	if err != gorm.ErrRecordNotFound {
+		return nil, err
+	}
+
+	// Create new user with unique alias
+	alias, err := generateUniqueAlias(db)
+	if err != nil {
+		return nil, err
+	}
+
+	user = &User{
+		ID:       id,
+		Username: username,
+		Alias:    alias,
+	}
+
+	if err := db.Create(user).Error; err != nil {
+		return nil, err
+	}
+	return user, nil
+}
+
+func generateUniqueAlias(db *gorm.DB) (string, error) {
+	// Generate cryptographically secure random index
+	nameIndex, err := rand.Int(rand.Reader, big.NewInt(int64(len(names))))
+	if err != nil {
+		return "", fmt.Errorf("failed to generate random name index: %w", err)
+	}
+
+	name := names[nameIndex.Int64()]
+	var lastUser User
+	pattern := fmt.Sprintf("%s%%", name)
+	result := db.Where("alias LIKE ?", pattern).Order("created_at DESC").First(&lastUser)
+
+	nextNum := 1
+	if result.Error == nil {
+		re := regexp.MustCompile(fmt.Sprintf(`^%s(\d+)$`, name))
+		if matches := re.FindStringSubmatch(lastUser.Alias); len(matches) == 2 {
+			if n, err := strconv.Atoi(matches[1]); err == nil {
+				nextNum = n + 1
+			}
+		}
+	}
```

Comment:
Non mi sembra che venga gestito da nessuna parte il caso in cui il DB dia errore. Può dare errore di RecordNotFound, però può anche produrre altri errori. Direi che un check è d'obbligo